packages/cli-kit/src/public/node/cli.ts

```diff
@@ -15,11 +15,22 @@ interface RunCLIOptions {
   development: boolean
 }
 
-async function exitIfOldNodeVersion(versions: NodeJS.ProcessVersions = process.versions) {
+const DEFAULT_MIN_NODE_VERSION = 18
+
+async function exitIfOldNodeVersion(moduleURL: string, versions: NodeJS.ProcessVersions = process.versions) {
   const nodeVersion = versions.node
   const nodeMajorVersion = Number(nodeVersion.split('.')[0])
 
-  const currentSupportedNodeVersion = 18
+  const {findUpAndReadPackageJson} = await import('./node-package-manager.js')
+  const {moduleDirectory} = await import('./path.js')
+
+  const packageJson = await findUpAndReadPackageJson(moduleDirectory(moduleURL))
+  const engines = (packageJson.content as {engines?: {node?: string}}).engines
+  const nodeRequirement = engines?.node ?? `>=${DEFAULT_MIN_NODE_VERSION}.0.0`
```

Comment:
`engines.node` should always exist in cli-kit's package.json, so I'd prefer to just log an error and return when undefined. 

Otherwise, if for some reason this is not working as expected, we would be checking against a different minimum node version here that it's outdated.

Reply:
Agree with Gonzalo, but also remember that cli-kit is bundled, we should test that this works on a snapshot too.
The package.json that this code will read in the released CLI will be the one under `packages/cli`